django/utils/http.py

```diff
@@ -88,11 +88,21 @@ def urlencode(query, doseq=False):
         query = query.lists()
     elif hasattr(query, 'items'):
         query = query.items()
-    return original_urlencode(
-        [(k, [str(i) for i in v] if isinstance(v, (list, tuple)) else str(v))
-         for k, v in query],
-        doseq
-    )
+
+    query_params = []
+    for key, value in query:
+        if isinstance(value, (list, tuple)):
+            query_val = [
+                item if isinstance(item, bytes) else str(item)
+                for item in value
+            ]
+        elif isinstance(value, bytes):
+            query_val = value
+        else:
+            query_val = str(value)
```

Comment:
As the Python `urlencode()` already handles scalars, I think this could be slightly simplified to:

```py
        if isinstance(value, (list, tuple)):
            query_val = [
                item if isinstance(item, bytes) else str(item)
                for item in value
            ]
        else:
            query_val = value
```